lld/test/MachO/objc-relative-method-lists-simple.s

```diff
@@ -2,15 +2,15 @@
 # UNSUPPORTED: target=arm{{.*}}-unknown-linux-gnueabihf
```

Comment:
We need to add cases to test the new logic. Specifically: 

MacOS 10 => defaults to not relative
MacOS 11 => defaults to relative 

MacOS 10 + `-objc_relative_method_lists` => relative
MacOS 12 + `-no_objc_relative_method_lists`=> not relative

Reply:
1. MacOS 10 => defaults to not relative
2. MacOS 11 => defaults to relative
3. MacOS 10 + -objc_relative_method_lists => relative
4. MacOS 12 + -no_objc_relative_method_lists=> not relative

2. Should be covered by removing `-objc_relative_method_lists` in the tests 
4. Should be covered by unchanged `-no_objc_relative_method_lists` on [Line 17](https://github.com/DataCorrupted/llvm-project/blob/HEAD/lld/test/MachO/objc-relative-method-lists-simple.s#L17C1-L17C135) (on MacOS 11 tho)

I need to modify lit config to test 1 and 3 (our lit config defaults to MacOS 11 now). I tried yesterday but with no luck. Let me try it again.

Reply:
> our lit config defaults to MacOS 11 now

Why change lit configs - why not manually overwrite on the command line ? Like [here](https://github.com/llvm/llvm-project/blob/38ef6929a3322fdddd74b3d6abdf6936cc4d8e62/lld/test/MachO/invalid/incompatible-arch.s#L19).